### ISSUE TITLE
[Bug]: Eqnarray in AnchoredText results in misplaced text (new in v3.6.0)

### ISSUE THREAD

**Issue by BogdanRajkov** [matplotlib/matplotlib#23977]
### Bug summary

The title pretty much sums it up. I have no idea if this only happens for the eqnarray environment and AnchoredText or for all Latex text and other boxes, e.g. legends, but also don't have the time to test it myself. I'll simply keep using v3.5.3, but I hope this helps other people get to the root cause.

### Code for reproduction

```python
import numpy as np
import matplotlib.pyplot as plt
import matplotlib
from matplotlib.offsetbox import AnchoredText

v = matplotlib.__version__
fig, ax = plt.subplots()
x = np.linspace(0, 2*np.pi)
y = np.sin(x)
plt.plot(x, y)

text = ('\\begin{eqnarray*}'
        f'foo \\\\'
        f'{v=} \\\\'
        '\\end{eqnarray*}')
at = AnchoredText(text, loc='upper left', prop={'usetex': True})
# at = AnchoredText(text, loc='upper left')
ax.add_artist(at)
plt.show()
```

### Actual outcome

![new_behaviour](https://user-images.githubusercontent.com/33605382/191560351-79526627-c48a-4fa5-85ae-f3bd1f4fd1ee.png)

### Expected outcome

![old_behaviour](https://user-images.githubusercontent.com/33605382/191560387-54ef0698-f87f-4be0-8068-92998e2c96aa.png)

### Additional information

I should mention that when `usetex` is set to False, the text is displayed just fine (by text I mean the code, obviously it doesn't compile), so I can narrow it down at least that much.

### Operating system

Windows 11

### Matplotlib Version

3.6.0

### Matplotlib Backend

QtAgg

### Python version

Python 3.10.1

### Installation

pip

**Comment by dstansby** [comment]
Looking into this now - bisects to c973552b49d6d6c583a52d1ac838bc0748462666

**Comment by dstansby** [comment]
Had a quick look and can't work out what the fix is - perhaps @anntzer will have more luck as you wrote the failing commit.

**Comment by anntzer** [comment]
I can confirm the issue and repro it even more simply with e.g.
```python
import matplotlib.pyplot as plt
plt.text(.5, .5, r'$$E = mc^2$$', usetex=True)
plt.axvline(.5)
plt.axhline(.5)
plt.show()
```
the basic problem being that we don't get a baseline mark *at the baseline of the last line of displaymath mode*, but only at the baseline of the (empty) following paragraph.

Probably there's some solution which involves fiddling with TeX internals (and I still think the general approach is the correct one), but it doesn't look obvious to fix, so possibly in the short term #22360 should be reverted for now and thus also #22361 (first), and #22350 (which was the original issue that triggered these changes) reopened.

**Comment by anntzer** [comment]
Actually I figured out a solution, see #24011.

**Comment by QuLogic** [comment]
Should be fixed by the revert in #24047.
